common/test/client.go

```diff
@@ -82,7 +82,9 @@ func (r *Request) Do() (err error) {
 	if err != nil {
 		return err
 	}
-	defer (func() { err = res.Body.Close() })()
+	defer (func() { 
+		if err == nil { err = res.Body.Close() }
```

Comment:
If `err != nil`, the body won't be closed since `res.Body.Close` won't be called.